Application/EdFi.Ods.Api/Extensions/HttpRequestExtensions.cs

```diff
@@ -43,45 +43,56 @@ public static string Scheme(this HttpRequest request, bool useProxyHeaders = fal
                 scheme = proxyHeaderValue;
             }
 
-            return scheme;
+            // The x-forwarded-proto header can contain a single originating protocol or, in some 
+            // cases, multiple protocols. Seee ODS-5481 for more information. We care about the
+            // _first_ protocol listed.
+            if (scheme == null) { return "http"; }
+
+            return scheme.Split(',')[0];
         }
 
-        public static string Host(this HttpRequest request, bool useProxyHeaders = false)
+        public static string Host(this HttpRequest request, bool useProxyHeaders = false, string defaultForwardedHostServer = "localhost")
         {
-            string host = request.Host.Host;
-
+            // Use actual request host when not configured for use behind a reverse proxy
             if (!useProxyHeaders)
             {
-                return host;
+                return request.Host.Host;
             }
 
+            // Try to extract a X-Forwarded-Host value
             request.TryGetRequestHeader(HeaderConstants.XForwardedHost, out string proxyHeaderValue);
-
-            // Pass through for any value provided, null means header wasn't found so default to request
-            if (proxyHeaderValue != null)
+            if (!string.IsNullOrWhiteSpace(proxyHeaderValue))
             {
-                host = proxyHeaderValue;
+                return proxyHeaderValue;
             }
 
-            return host;
+            // Fallback to appsettings value, if available
+            return defaultForwardedHostServer;
         }
 
-        public static int Port(this HttpRequest request, bool useProxyHeaders = false)
+        public static int Port(this HttpRequest request, bool useProxyHeaders = false, int defaultForwardedHostPort = 443)
         {
-            var defaultPortForScheme = Scheme(request, useProxyHeaders) == "https"
-                ? 443
-                : 80;
-
+            // User actual request host when not configured for use behind a reverse proxy
             if (!useProxyHeaders)
             {
-                return request.Host.Port ?? defaultPortForScheme;
+                return request.Host.Port ?? getDefaultPort();
             }
 
+            // Try to extract a X-Forwarded-Port value
             request.TryGetRequestHeader(HeaderConstants.XForwardedPort, out string proxyHeaderValue);
 
-            return !int.TryParse(proxyHeaderValue, out int port)
-                ? request.Host.Port ?? defaultPortForScheme
-                : port;
+            if (proxyHeaderValue != null)
```

Comment:
Should we check for ``IsNullOrWhiteSpace`` here as well? To handle the scenario where the header is defined with an empty value.

Reply:
Possibly. Will look into it.

Reply:
This does behave correctly in that scenario. I've added Postman tests and covered this in unit tests as well.

Reply:
done